source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -747,35 +747,42 @@ namespace
     template<class T>
     void createInterface(const ElementPtr& src, T schema)
     {
-        for (auto elem : src->getChildrenOfType<ValueElement>())
+        StringSet elementsAdded;
+        for (ConstElementPtr interface : src->traverseInheritance())
```

Comment:
There was bug in the interface building code. It did not handle inheritance at all for nodedefs.